Network/MainClient.py

```diff
@@ -0,0 +1,161 @@
+import socket
+import threading
+import keyboard
+import time
+import sys
+import math
+
+import pygame
+
+pygame.init()
+
+font = pygame.font.Font(None, 50)
+
+screen_width = 800
+screen_height = 400
+screen = pygame.display.set_mode((screen_width, screen_height))
+
+class paddle:
+    def __init__(self, x, y, width, height, colour):
+        self.y = y
+        self.x = x
+        self.width = width
+        self.height = height
+        self.score = 0
+        self.speed = 3
+        self.image = pygame.Surface((width, height))
+        self.image.fill(colour)
+        self.colour = colour
+
+    def draw(self):
+        screen.blit(self.image, (self.x, self.y))
+
+class Ball:
+    def __init__(self, x, y, width, height, direction):
+        self.starting_pos = (x, y)
+        self.x = x
+        self.y = y
+        self.width = width
+        self.height = height
+        self.speed = 5
+        self.direction = direction
+        self.image = pygame.Surface((width, height))
+        self.image.fill((255, 255, 255))
+
+    def draw(self):
+        screen.blit(self.image, (self.x, self.y))
+
+player1 = paddle(10, 150, 11, 100, (0, 255, 0))
+player2 = paddle(780, 150, 11, 100, (255, 0, 0))
+ball = Ball(400, 200, 20, 20, 45)
+
+class Client:
+    def __init__(self, host='localhost', port=12345):
+        self.host = host
+        self.port = port
+        self.sock = socket.socket()
+        self.stopflag = False
+
+        # Game state variables
+        self.p1x = 10
```

Comment:
By storing the game state here and in the `Paddle`/`Ball` entities, you've got two sources of truth for the game world. Could you move the player paddles and ball into the class, and remove all of these except the score variables?